apps/extension/src/background/keyring/keyring.ts

```diff
@@ -1,68 +1,217 @@
 import { KVStore } from "@anoma/storage";
+import { AEAD, Bip44, Mnemonic, PhraseSize } from "@anoma/crypto";
+import { Address } from "@anoma/shared";
+import { KeyRingStatus, MnemonicState, AccountState } from "./types";
+import { v5 as uuid } from "uuid";
+import { Bip44Path, DerivedAccount } from "./types";
 
-export enum KeyRingStatus {
-  NOTLOADED,
-  EMPTY,
-  LOCKED,
-  UNLOCKED,
+import { IStore, Store } from "../types";
+
+// Generated UUID namespace for uuid v5
+const UUID_NAMESPACE = "9bfceade-37fe-11ed-acc0-a3da3461b38c";
+
+// Construct unique uuid, passing in an arbitray number of arguments.
+// This could be a unique parameter of the object receiving the id,
+// or an index based on the number of existing objects in a hierarchy.
+const getId = (name: string, ...args: (number | string)[]): string => {
+  // Ensure a unique UUID
+  let uuidName = name;
+
+  // Concatenate any number of args onto the name parameter
+  args.forEach((arg) => {
+    uuidName = `${uuidName}::${arg}`;
+  });
+
+  return uuid(uuidName, UUID_NAMESPACE);
+};
+
+enum KeyRingStore {
+  Mnemonic = "mnemonic",
+  Account = "account",
 }
 
-/*
- Keyring stores keys in persistent backround.
+/**
+ * Keyring stores keys in persisted backround.
  */
 export class KeyRing {
-  private _loaded: boolean;
+  private _mnemonicStore: IStore<MnemonicState>;
+  private _accountStore: IStore<AccountState>;
+  private _password: string | undefined;
```

Comment:
Nit: I assume that the underscores are in all of these just for the consistency, while we actually only need it here for `status` which has a getter? I see this also in other places. Best we decide a consistent rule.

Reply:
The `_` prefix is for any private variables, which I have been doing consistently. I see different approaches across other extensions and wallet apps, and it just comes down to what we want to do for consistency. Some are very strict about prefixing `_` to _any_ private variable (See polkadot-js), whereas I rarely see that done in Keplr. I think scoping certain things as private and indicating it as such is important, but that's just an opinion. I think it's better for the developer (or anyone new to the code) to know that a method or value is only accessible from inside a class, and it's a very common convention. I also think it's good form to scope these things with intention, only allowing public access to public values, etc., otherwise it just gets sloppy and hard to reason about. I like the Rust approach, where everything is private by default, and you must be explicit about making something public (which is the opposite of TS/JS, where everything is public by default), but that's an opinion, and we should determine what works best for us!